gcloud/logging/connection.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Create / interact with gcloud logging connections."""
+
+from gcloud import connection as base_connection
+
+
+class Connection(base_connection.JSONConnection):
+    """A connection to Google Cloud Pubsub via the JSON REST API.
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials`
+    :param credentials: (Optional) The OAuth2 Credentials to use for this
+                        connection.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: (Optional) HTTP object to make requests.
+
+    :type api_base_url: string
+    :param api_base_url: The base of the API call URL. Defaults to the value
+                         :attr:`Connection.API_BASE_URL`.
+    """
+
+    API_BASE_URL = 'https://logging.googleapis.com'
+    """The base of the API call URL."""
+
+    API_VERSION = 'v2beta1'
+    """The version of the API, used in building the API call's URL."""
+
+    API_URL_TEMPLATE = '{api_base_url}/{api_version}{path}'
+    """A template for the URL of a particular API call."""
+
+    SCOPE = ('https://www.googleapis.com/auth/logging.read',
+             'https://www.googleapis.com/auth/logging.write',
+             'https://www.googleapis.com/auth/logging.admin',
+             'https://www.googleapis.com/auth/cloud-platform')
+    """The scopes required for authenticating as a Cloud Pub/Sub consumer."""
+
+    def __init__(self, credentials=None, http=None, api_base_url=None):
+        super(Connection, self).__init__(credentials=credentials, http=http)
+        if api_base_url is None:
+            api_base_url = self.__class__.API_BASE_URL
+        self.api_base_url = api_base_url
+
+    def build_api_url(self, path, query_params=None,
+                      api_base_url=None, api_version=None):
```

Comment:
Might be an artifact from Pub/Sub's version, which had a bunch of stuff in place to support the emulator.